crates/noirc_frontend/src/hir_def/types.rs

```diff
@@ -939,91 +664,65 @@ impl Type {
                             *binding.borrow_mut() = TypeBinding::Bound(clone);
                             Ok(())
                         }
-                        TypeVariableKind::Constant(_) | TypeVariableKind::IntegerOrField(_) => {
-                            Err(SpanKind::None)
+                        TypeVariableKind::Constant(_) | TypeVariableKind::IntegerOrField => {
+                            Err(UnificationError)
                         }
                     },
                 }
             }
-            _ => Err(SpanKind::None),
+            _ => Err(UnificationError),
         }
     }
 
     /// Try to bind a PolymorphicInt variable to self, succeeding if self is an integer, field,
     /// other PolymorphicInt type, or type variable. If use_subtype is true, the CompTime fields
```

Comment:
This still mentions `CompTime`

Reply:
I went through and removed all comment references and a few errors that aren't produced anymore